soc/ite/ec/it51xxx/chip_chipregs.h

```diff
@@ -307,6 +307,8 @@ struct gctrl_it51xxx_regs {
 #define IT51XXX_GCTRL_LRSIPGWR    BIT(0)
 /* 0x38: Special Control 9 */
 #define IT51XXX_GCTRL_ALTIE       BIT(4)
+/* 0x47: Scratch SRAM0 Base Address */
+#define IT51XXX_SEL_SRAM0_BASE_4K 0x04
```

Comment:
Is there a register encoding happening here?  How does the value `0x04` set the SRAM base address to 0x801000?

Reply:
Yes, there is register encoding. The `GCTRL_SCR0BAR `register uses a 4KB unit offset encoding.
As mentioned in the comment above, setting the value to 0x04 configures the SRAM base address to 0x801000.